docs/SPIRVRepresentationInLLVM.rst

```diff
@@ -213,6 +232,48 @@ following format:
   !<InstructionMetadata1> = !{<Operand1>, <Operand2>, ..}
   !<InstructionMetadata2> = !{<Operand1>, <Operand2>, ..}
 
++--------------------+---------------------------------------------------------+
+| SPIR-V instruction | LLVM IR                                                 |
++====================+=========================================================+
+| OpSource           | .. code-block:: llvm                                    |
+|                    |                                                         |
+|                    |    !spirv.Source = !{!0}                                |
+|                    |    !0 = !{i32 3, i32 102000, !1}                        |
```

Comment:
Could I ask where the numbering of OpenCL C versions for SPIR-V is specified?

I can see that it's defined within the SPIR-V/LLVM translator in lib/SPIRV/OCLUtil.h as follows:
```
const unsigned CL12 = 102000;
const unsigned CL20 = 200000;
const unsigned CL21 = 201000;
const unsigned CL30 = 300000;
```
However, that doesn't tell me why.

I find this confusing, as I would expect -cl-std=CL1.2 to be 120 or 10200, and not 102000.

The SPIR-V spec gives `OpSource GLSL 450` as an example, and the tests in SPIR-V Tools include e.g. `OpSource OpenCL_C 120`, `OpSource OpenCL_C 12` and ``OpSource OpenCL_C 1202000`.

Note that the SPIR-V file header encodes the SPIR-V version as bytes, e.g. 0x00010300 for SPIR-V 1.3, but this is the _SPIR-V_ version, and it is encoded with a _byte per component_, and not as a pair of decimal digits per component, and there isn't the extra trailing zero.

My concern here is whether we want to standardize what seems to be a bug.

Reply:
> Could I ask where the numbering of OpenCL C versions for SPIR-V is specified?

https://www.khronos.org/registry/OpenCL/specs/2.2/html/OpenCL_Env.html#_source_language_encoding

I think I got this example from one of tests, so there might be bugs. Probably the format provided there is quite outdated.
If I understand OpenCL environment specification correctly, the 32-bit literal consists of 4 bytes: 0 | Major Number | Minor Number | Revision Number (optional). 
OpenCL 1.2 -> hex (0x00010200) -> dec 66048.

```suggestion
|                    |    !0 = !{i32 3, i32 66048, !1}                        |
```
Right?

Reply:
@StuartDBrady, are you okay to commit the change suggested in my previous comment?

Reply:
The suggestion looks fine to me.

Reply:
Thanks. I've applied the change.